jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.cpp

```diff
@@ -113,24 +113,47 @@ ecma_builtin_array_prototype_object_concat (ecma_value_t this_arg, /**< this arg
   ecma_completion_value_t new_array = ecma_op_create_array_object (0, 0, false);
   ecma_object_t *new_array_p = ecma_get_object_from_completion_value (new_array);
 
+  /* 5.b */
+  if (ecma_object_get_class_name (obj_p) == LIT_MAGIC_STRING_ARRAY_UL)
```

Comment:
Seems that the block is very similar to code in the loop below. Could we could move this to a helper and call from both places?